source/postprocess/gravity_point_values.cc

```diff
@@ -0,0 +1,322 @@
+          model_inner_radius = dynamic_cast<const GeometryModel::SphericalShell<dim>&>
+                               (this->get_geometry_model()).inner_radius();
+        }
+      else if (dynamic_cast<const GeometryModel::Chunk<dim>*> (&this->get_geometry_model()) != 0)
+        {
+          model_outer_radius = dynamic_cast<const GeometryModel::Chunk<dim>&>
+                               (this->get_geometry_model()).outer_radius();
+          model_inner_radius = dynamic_cast<const GeometryModel::Chunk<dim>&>
+                               (this->get_geometry_model()).inner_radius();
+        }
+      else if (dynamic_cast<const GeometryModel::Sphere<dim>*> (&this->get_geometry_model()) != 0)
+        {
+          model_outer_radius = dynamic_cast<const GeometryModel::Sphere<dim>&>
+                               (this->get_geometry_model()).radius();
+          model_inner_radius = 0;
+        }
+      else
+        {
+          Assert (false, ExcMessage ("This initial condition can only be used if the geometry "
+                                     "is a sphere, a spherical shell, a chunk or an "
+                                     "ellipsoidal chunk."));
+          model_outer_radius = 1;
+          model_inner_radius = 0;
+        }
+
+      // Get the value of the universal gravitational constant
+      const double G = aspect::constants::big_g;
+
+      // now write all data to the file of choice. start with a pre-amble that
+      // explains the meaning of the various fields
+      const std::string filename = (this->get_output_directory() +
+                                    "output_gravity");
+      std::ofstream f (filename.c_str());
+      f << "# 1: position_satellite_r" << '\n'
+        << "# 2: position_satellite_phi" << '\n'
+        << "# 3: position_satellite_theta" << '\n'
+        << "# 4: position_satellite_x" << '\n'
+        << "# 5: position_satellite_y" << '\n'
+        << "# 6: position_satellite_z" << '\n'
+        << "# 7: gravity_x" << '\n'
+        << "# 8: gravity_y" << '\n'
+        << "# 9: gravity_z" << '\n'
+        << "# 10: gravity_norm" << '\n'
+        << "# 11: gravity_theory" << '\n'
+        << "# 12: gravity_diff (theoric-norm)" << '\n'
+        << "# 13: potential" << '\n'
+        << '\n';
+
+      // Temporary output during the build of this postprocessor WIP
+      const std::string filename2 = (this->get_output_directory() +
+                                     "checks_wip.txt");
+      std::ofstream f2 (filename2.c_str());
+      f2 << minimum_radius << ' ' << maximum_radius << '\n'
+         << minimum_longitude << ' ' << maximum_longitude << '\n'
+         << minimum_latitude << ' ' << maximum_latitude << '\n';
+
+      // Storing cartesian coordinate, density and JzW at quadrature points in a vector
+      // avoids to use MaterialModel and fe_values within the loops.
+      unsigned int local_cell_number = (this->get_triangulation().n_locally_owned_active_cells());
```

Comment:
make this `const` and use a different variable as loop counter below, currently you use the same variable for different purposes. Also generally we name these variables `n_locally_owned_cells` not `locally_owned_cell_number`.